pybamm/solvers/casadi_solver.py

```diff
@@ -143,6 +145,33 @@ def _integrate(self, model, t_eval, inputs=None):
                     [event(t, y0, inputs) for event in model.terminate_events_eval]
                 )
             )
+
+            extrap_event = [
```

Comment:
do we need to check the initial conditions? I guess they could lead to extrapolation, but it would be nice to not repeat this code

Reply:
I did it so it fails faster when you give an initial condition that extrapolates. What do you prefer?

Reply:
yeah probably better to fail faster

Reply:
I will make this sentence my motto from now on haha